zerver/openapi/zulip.yaml

```diff
@@ -5265,6 +5254,131 @@ paths:
+        - name: type
+          in: query
+          description: |
+            The type of scheduled message to be sent. `"direct"` for a direct
+            message and `"stream"` for a stream message.
+
+            When updating the type of the scheduled message, the `to` parameter
+            is required. And, if updating the type of the scheduled message to
+            `"stream"`, then the `topic` parameter is also required.
+
+            In Zulip 7.0 (feature level 174), `"direct"` was added as the
+            preferred way to indicate the type of a direct message, deprecating
+            the original `"private"`. While `"private"` is supported for
+            scheduling direct messages, clients are encouraged to use to the
+            modern convention because support for `"private"` may eventually
+            be removed.
+          schema:
+            type: string
+            enum:
+              - direct
+              - stream
+              - private
+          example: stream
+        - name: to
+          in: query
+          description: |
+            The scheduled message's tentative target audience.
+
+            For stream messages, the integer ID of the stream.
+            For direct messages, a list containing integer user IDs.
+
+            Required when updating the `type` of the scheduled message.
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - type: integer
+                  - type: array
+                    items:
+                      type: integer
+                    minLength: 1
+              example: 8
+        - name: content
+          in: query
+          description: |
+            The updated content of the scheduled message.
+
+            Clients should use the `max_message_length` returned by the
+            [`POST /register`](/api/register-queue) endpoint to determine
+            the maximum message size.
+          schema:
+            type: string
+          example: Hello
+        - name: topic
+          in: query
+          description: |
+            The updated topic of the scheduled message.
+
+            Required when updating the `type` of the scheduled message to
+            `"stream"`.
+
+            Clients should use the `max_topic_length` returned by the
+            [`POST /register`](/api/register-queue) endpoint to determine
+            the maximum topic length.
+          schema:
+            type: string
+          example: Castle
+        - name: scheduled_delivery_timestamp
+          in: query
+          schema:
+            type: integer
+          description: |
+            The UNIX timestamp for when the message will be sent,
+            in UTC seconds.
+
+            Required when updating a scheduled message that the server
+            failed to send. See API changelog feature level 181 note for
+            more information on identifying scheduled messages that failed.
```

Comment:
This reference should come with some kind of link (even if words are required to point more specifically to a part of a page that doesn't have its own anchor).

My first reading was actually that this meant some "Changes" note on this page; it was only after not finding one of those for 181 that I realized it meant the changelog at https://zulip.com/api/changelog . We don't currently have links for individual feature levels (though adding those would be a nice touch), but (the relativized version of) https://zulip.com/api/changelog#changes-in-zulip-70 would work well.

---

Separately, I feel like in principle this shouldn't involve referring to any kind of changelog item at all — it's describing the current API, not the differences in any previous API, so it should be self-contained within the description of the current API. If there isn't a good thing to point to in the non-changelog parts of the docs, then that's a sign we should add something.

Here's one version, though:
```suggestion
            failed to send. This state is indicated with `"failed": true` in scheduled-message objects;
            see description at [`GET /scheduled_messages`](/api/get-scheduled-messages).
```
That version leaves implicit that it applies to a "scheduled-message object" as seen in the event system too; alternatively that could be made explicit with some more words.